deepdrivemd/misc/aggregator_reader.py

```diff
@@ -0,0 +1,193 @@
+        self.io = self.adios.DeclareIO(stream_name)
+        self.stream = self.io.Open(fn, adios2.Mode.Read)
+
+    def __del__(self):
+        self.stream.Close()
+        self.io.RemoveAllVariables()
+        self.adios.RemoveAllIOs()
+
+    def next_all(self, N):
+        CMs = []
+        POSITIONs = []
+        MD5s = []
+        STEPs = []
+        RMSDs = []
+        VELOCITYs = []
+        MD5s = []
+        for i in range(N):
+            status = self.stream.BeginStep(adios2.StepMode.Read, 0.0)
+            if status != adios2.StepStatus.OK:
+                break
+
+            step = np.zeros(1, dtype=np.int32)
+            varStep = self.io.InquireVariable("step")
+            self.stream.Get(varStep, step)
+
+            rmsd = np.zeros(1, dtype=np.float32)
+            varRMSD = self.io.InquireVariable("rmsd")
+            self.stream.Get(varRMSD, rmsd)
+
+            varCM = self.io.InquireVariable("contact_map")
+            shapeCM = varCM.Shape()
+            ndimCM = len(shapeCM)
+            start = [0] * ndimCM
+            count = shapeCM
+            varCM.SetSelection([start, count])
+            cm = np.zeros(shapeCM, dtype=np.uint8)
+            self.stream.Get(varCM, cm)
+
+            varPositions = self.io.InquireVariable("positions")
+            shapePositions = varPositions.Shape()
+            ndimPositions = len(shapePositions)
+            start = [0] * ndimPositions
+            count = shapePositions
+            varPositions.SetSelection([start, count])
+            positions = np.zeros(shapePositions, dtype=np.float32)
+            self.stream.Get(varPositions, positions)
+
+            varVelocities = self.io.InquireVariable("velocities")
+            shapeVelocities = varVelocities.Shape()
+            ndimVelocities = len(shapeVelocities)
+            start = [0] * ndimVelocities
+            count = shapeVelocities
+            varVelocities.SetSelection([start, count])
+            velocities = np.zeros(shapeVelocities, dtype=np.float32)
+            self.stream.Get(varVelocities, velocities)
+
+            varMD5 = self.io.InquireVariable("md5")
+            shapeMD5 = varMD5.Shape()
+            ndimMD5 = len(shapeMD5)
+            start = [0] * ndimMD5
+            count = shapeMD5
+            md5 = self.stream.Get(varMD5)
+
+            self.stream.EndStep()
+
+            cm = np.unpackbits(cm)
+            cm = t1Dto2D(cm)
+
+            MD5s.append(md5)
+            CMs.append(cm)
+            STEPs.append(step[0])
+            RMSDs.append(rmsd[0])
+            POSITIONs.append(positions)
+            VELOCITYs.append(velocities)
+
+        return i, STEPs, MD5s, CMs, POSITIONs, VELOCITYs, RMSDs
+
+    def next_cm(self, N):
+        CMs = []
+        for i in range(N):
+            status = self.stream.BeginStep(adios2.StepMode.Read, 0.0)
+            if status != adios2.StepStatus.OK:
+                break
+
+            varCM = self.io.InquireVariable("contact_map")
```

Comment:
Use get_array function here to avoid code duplication.

Reply:
Handled by the new design